pkg/pubsub/pubsub.go

```diff
@@ -1,8 +1,70 @@
 package pubsub
 
-import "context"
+import (
+	"context"
+	"encoding/json"
+
+	"cloud.google.com/go/pubsub"
+	"go.uber.org/zap"
+)
 
 // Publisher ...
 type Publisher interface {
 	Publish(ctx context.Context, topic string, data interface{}) error
 }
+
+type PubsubClient struct {
+	client      *pubsub.Client
+	orderingKey string
+	logger      *zap.SugaredLogger
+}
+
+func NewPublisher(ctx context.Context, projectID string, orderingKey string) (*PubsubClient, error) {
```

Comment:
```suggestion
func NewPubsubClient(ctx context.Context, projectID string, orderingKey string) (*PubsubClient, error) {
```

We might have some other implementations such as NewKafkaClient, NewNatsClient, ...